### ISSUE TITLE
False positive: undefined-variable for return type

### ISSUE THREAD

**Issue by samuela** [PyCQA/pylint#2973]
### Steps to reproduce
```python
class Foo(NamedTuple):
  def add(self) -> Foo:
    return Foo()
```

### Current behavior
```
Undefined variable 'Foo'pylint(undefined-variable)
```
on the `-> Foo` type annotation.

### Expected behavior
No error since Foo has been defined and is a valid type.

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 27 2019, 09:23:15) 
[Clang 10.0.1 (clang-1001.0.46.3)]
```

**Comment by PCManticore** [comment]
Unless you use `from __future__ import annotations` this code is going to raise a `NameError` at runtime, so the check here is correct.
